packages/protocol/contracts/common/linkedlists/SortedLinkedList.sol

```diff
@@ -71,8 +71,18 @@ library SortedLinkedList {
     bytes32 lesserKey,
     bytes32 greaterKey
   ) internal {
-    remove(list, key);
-    insert(list, key, value, lesserKey, greaterKey);
+    require(contains(list, key), "key not in list");
+    require(key != bytes32(0) && key != lesserKey && key != greaterKey, "invalid key");
```

Comment:
we could switch these requirement since if the key is invalid it doesn't make sense to check if it is in the list which could safe a bit of gas

Reply:
Gas savings don't matter as much for failing transactions. I don't feel strongly, but I slightly prefer this ordering (it follows the order of checks in the old update method: first `contains` in `remove`, then validity of the key in `insert`).

Reply:
ok, consistency makes sense :-)